src/main/groovy/com/bmuschko/gradle/docker/utils/DockerThreadContextClassLoader.groovy

```diff
@@ -29,22 +29,23 @@ class DockerThreadContextClassLoader implements ThreadContextClassLoader {
     public static final String MODEL_PACKAGE = 'com.github.dockerjava.api.model'
     public static final String COMMAND_PACKAGE = 'com.github.dockerjava.core.command'
 
+    private final DockerExtension dockerExtension
+    private final Set<File> classpath
+    private def dockerClient
+
+    public DockerThreadContextClassLoader(DockerExtension dockerExtension, Set<File> classpath) {
+        this.dockerExtension = dockerExtension
+        this.classpath = classpath
+    }
+
     /**
      * {@inheritDoc}
      */
     @Override
-    void withClasspath(Set<File> classpathFiles, DockerClientConfiguration dockerClientConfiguration, Closure closure) {
-        ClassLoader originalClassLoader = getClass().classLoader
-
-        try {
-            Thread.currentThread().contextClassLoader = createClassLoader(classpathFiles)
-            closure.resolveStrategy = Closure.DELEGATE_FIRST
-            closure.delegate = this
-            closure(getDockerClient(dockerClientConfiguration))
-        }
-        finally {
-            Thread.currentThread().contextClassLoader = originalClassLoader
-        }
+    void withClosure(Closure closure) {
```

Comment:
The method name is not very speaking. I don't know what the closure works on or what it really does without looking at the implementation.